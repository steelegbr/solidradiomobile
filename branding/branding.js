/**
 * Provides branding functions.
 */

import { DefaultTheme, DarkTheme } from 'react-native-paper';

/**
 * Generates a station theme.
 * @param {bool} darkMode An indication if we are in dark mode.
 * @param {station} station The station this theme is for.
 */

export function generateStationTheme(darkMode, station) {

    let theme = generateTheme(darkMode);
    theme.colors.primary = station.primary_colour;
    theme.colors.accent = station.primary_colour;
    theme.mode = 'adaptive';
    return theme;

}

/**
 * Generates a theme with no station.
 * @param {bool} darkMode An indication if we are in dark mode.
 */

export function generateTheme(darkMode) {

    if (darkMode) {
        return {
            ...DarkTheme,
            roundness: 10,
            colors: {
                ...DarkTheme.colors
            }
        };
    } else {
        return {
            ...DefaultTheme,
            roundness: 10,
            colors: {
                ...DefaultTheme.colors
            }
        };
    }

}
