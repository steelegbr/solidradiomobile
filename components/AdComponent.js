import React from 'react';
import { Component } from "react";
import { connect } from "react-redux";
import { BannerAd, BannerAdSize, TestIds, AdsConsentStatus } from "@react-native-firebase/admob";

class AdComponent extends Component {

    render() {

        const { nonPersonalised, unitId } = this.props;

        if (__DEV__) {
            return(
                <BannerAd 
                    unitId={TestIds.BANNER} 
                    size={BannerAdSize.SMART_BANNER}
                    requestOptions={{
                        requestNonPersonalizedAdsOnly: nonPersonalised
                    }}
                />
            );
        } else {
            return(
                <BannerAd 
                    unitId={unitId} 
                    size={BannerAdSize.SMART_BANNER}
                    requestOptions={{
                        requestNonPersonalizedAdsOnly: nonPersonalised
                    }}
                    onAdFailedToLoad={adFailedToLoad}
                />
            );
        }


    }

}

/**
 * Handles an advert failing to load.
 * @param {error} error The error loading the advert.
 */

function adFailedToLoad(error) {
    
    const { adLoadError, unitId } = self.props;
    adLoadError(error, unitId);

}

function mapStateToProps(state, ownProps) {
    return {
        nonPersonalised: state.admob.consent == AdsConsentStatus.NON_PERSONALIZED,
        unitId: state.admob.units[ownProps.unitId]
    };
}

const mapDispatchToProps = dispatch => {
    return {
        adLoadError: (error, unitId) => dispatch(adLoadError(error, unitId))
    };
};

export default connect(mapStateToProps, mapDispatchToProps)(AdComponent);