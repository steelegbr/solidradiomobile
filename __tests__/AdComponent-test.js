/**
 * Tests ad component rendering.
 */

import React from 'react';
import AdComponent, { adFailedToLoad } from '../components/AdComponent';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import { generateTheme } from '../branding/branding';

// Firebase

jest.mock("@react-native-firebase/admob", () => ({
    AdsConsentStatus: {
        UNKNOWN: 0,
        NON_PERSONALIZED: 1,
        PERSONALISED: 2
    },
    TestIds: {
        BANNER: "bannerTestId"
    },
    BannerAdSize: {
        SMART_BANNER: "SMART_BANNER"
    },
    BannerAd: "BannerAd"
}));

// Store

const mockStore = configureStore([]);
let store;

beforeEach(() => {
    store = mockStore({
        admob: {
            consent: 2,
            units: {
                "testUnitName": "testUnitId"
            }
        },
        theme: generateTheme(false)
    });
});

// Tests

it('renders-correctly', () => {

    // Arrange & Act

    const ad = renderer.create(
        <Provider store={store}>
            <AdComponent unitId="testUnitName" />
        </Provider>
    ).toJSON();

    // Assert

    expect(ad).toMatchSnapshot();

});

it('renders-correctly-production', () => {

    // Arrange

    global.__DEV__ = false;

    // Act

    const ad = renderer.create(
        <Provider store={store}>
            <AdComponent unitId="testUnitName" />
        </Provider>
    ).toJSON();

    // Assert

    expect(ad).toMatchSnapshot();

});
