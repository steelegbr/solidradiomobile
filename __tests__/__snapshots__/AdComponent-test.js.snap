// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders-correctly 1`] = `
<BannerAd
  requestOptions={
    Object {
      "requestNonPersonalizedAdsOnly": false,
    }
  }
  size="SMART_BANNER"
  unitId="bannerTestId"
/>
`;
